File: src/ExamRegister.UI/src/app/shared/DTOs/paciente-dto.ts
import { HistoricopacienteDTO } from "./historicopaciente-dto";
import { PacienteinformacaoDTO } from "./pacienteinformacao-dto";

export interface PacienteDTO {
    idpaciente: string;
    idexterno: string;
    nome: string;
    sexo: string;
    nascimento: Date;
    inativo?: Date;
    historicoPaciente: HistoricopacienteDTO[];
    pacienteInformacao: PacienteinformacaoDTO[];
}

File: src/ExamRegister.WebApi.Abstractions/DTO/PacienteDTO.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace ExamRegister.WebApi.Abstractions.DTO
{
    public partial class PacienteDTO
    {
        public Guid idpaciente { get; set; }
        public string idexterno { get; set; }
        public string nome { get; set; }
        public char sexo { get; set; }
        public DateTime nascimento { get; set; }
        public DateTime? inativo { get; set; }

        public IEnumerable<HistoricoPacienteDTO> historicoPaciente { get; set; }
        public IEnumerable<PacienteInformacaoDTO> pacienteInformacao { get; set; }
    }
}
